### PR TITLE
[1840] fixed reassign

### LINKED ISSUE #6023: [1840] Warn when company's train ends when a train is not assigned
**What's your idea?**

Can you make a check to warn people when they are ending their turn with a tram that does not have a train assigned to it?

### LINKED ISSUE #6442: [1840] Remove the "Skip (Reassign)" trams button, Warn player if trams are unassigned
The tram reassignment interface has a redundant and confusing button to skip reassign, that is located right above the reassign trains button, making it easy to press by mistake. Simply do not have the option to skip, the player can press reassign trams without having changed anything to make no changes. With both buttons, it is possible for a player to think they have reassigned trams by using the toggles, then press skip (reassign) by mistake and have it undo their work. [Example ](https://18xx.games/game/58028?action=778) of this happening in game, and creating confusion since it was not discovered until the following OR.

As a fail safe, it would also be helpful to have a confirm button if a player ends the reassign trams step with trams unassigned. While there are rare circumstances where they may wish to do this intentionally, it is rare enough that an "are you sure" dialog of some kind would be helpful.

### PR DESCRIPTION
**Pull request by Zwergenpunk**
closes #6023
closes #6442

### CODE REVIEW

File: lib/engine/game/g_1840/step/reassign_trains.rb [hunk 1]
```diff
@@ -71,7 +69,16 @@ def process_reassign_trains(action)
                     "#{invalid_tram_corp.full_name} cannot be assigned more than one train"
             end
 
-            @log << "#{entity.owner.name} reassignes trains: #{reassignments.join(', ')}" unless reassignments.empty?
+            if entity.trains.length.positive? && @game.tram_owned_by_corporation[entity].any? { |item| item.trains.length.zero? }
```
Comment (tobymao) [inline lib/engine/game/g_1840/step/reassign_trains.rb:72]: size

File: lib/engine/game/g_1840/step/reassign_trains.rb [hunk 2]
```diff
@@ -71,7 +69,16 @@ def process_reassign_trains(action)
                     "#{invalid_tram_corp.full_name} cannot be assigned more than one train"
             end
 
-            @log << "#{entity.owner.name} reassignes trains: #{reassignments.join(', ')}" unless reassignments.empty?
+            if entity.trains.length.positive? && @game.tram_owned_by_corporation[entity].any? { |item| item.trains.size.zero? }
```
Comment (tobymao) [inline lib/engine/game/g_1840/step/reassign_trains.rb:72]: item.strains.empty?

!entity.trains.empty?

File: lib/engine/game/g_1840/step/reassign_trains.rb [hunk 3]
```diff
@@ -69,7 +69,7 @@ def process_reassign_trains(action)
                     "#{invalid_tram_corp.full_name} cannot be assigned more than one train"
             end
 
-            if entity.trains.size.positive? && !@game.tram_owned_by_corporation[entity].empty? { |item| item.trains.size.zero? }
+            if entity.trains.size.positive? && !@game.tram_owned_by_corporation[entity].empty? { |item| item.trains.empty? }
```
Comment (tobymao) [inline lib/engine/game/g_1840/step/reassign_trains.rb:72]: entity.trains.empty?

why size.positive?

Comment (Zwergenpunk) [reply lib/engine/game/g_1840/step/reassign_trains.rb:72]: needs to be >0 . what do you prefer?

Comment (Zwergenpunk) [reply lib/engine/game/g_1840/step/reassign_trains.rb:72]: i could just make !empty

**Review by tobymao — APPROVED** [review]

**Pull request merged**